src/serializer.test.ts

```diff
@@ -0,0 +1,472 @@
+    expect(children[0].text).toBe('Hello World');
+  });
+
+  it('deserializes an element with no children', () => {
+    const deserialized = deserializeHtmlString('<p/>');
+
+    expect(deserialized).toHaveLength(1);
+    expect(deserialized[0].type).toBe('paragraph');
+
+    const children = deserialized[0].children;
+    expect(children).toHaveLength(1);
+    expect(children[0].text).toBe('');
+  });
+
+  it('deserializes document with various headers and text tags', () => {
+    const deserialized = deserializeHtmlString(
+      '<h1>Header 1</h1><h2>Header 2</h2><h3>Header 3</h3><h4>Header 4</h4><h5>Header 5</h5><p><strong>bold, </strong><u>underlined</u> and <i>italic</i> text<code> + some code</code></p>'
+    );
+
+    expect(deserialized).toHaveLength(6);
+    expect(deserialized[0].type).toBe('h1');
+    expect(deserialized[1].type).toBe('h2');
+    expect(deserialized[2].type).toBe('h3');
+    expect(deserialized[3].type).toBe('h4');
+    expect(deserialized[4].type).toBe('h5');
+    expect(deserialized[5].type).toBe('paragraph');
+
+    expect(deserialized[0].children[0].text).toBe('Header 1');
+    expect(deserialized[1].children[0].text).toBe('Header 2');
+    expect(deserialized[2].children[0].text).toBe('Header 3');
+    expect(deserialized[3].children[0].text).toBe('Header 4');
+    expect(deserialized[4].children[0].text).toBe('Header 5');
+
+    const paragraph = deserialized[5];
+
+    expect(paragraph.children).toHaveLength(6);
+
+    expect(paragraph.children[0].text).toBe('bold, ');
+    checkMarks(paragraph.children[0], { bold: true });
+
+    expect(paragraph.children[1].text).toBe('underlined');
+    checkMarks(paragraph.children[1], { underline: true });
+
+    expect(paragraph.children[2].text).toBe(' and ');
+    checkMarks(paragraph.children[2], {});
+
+    expect(paragraph.children[3].text).toBe('italic');
+    checkMarks(paragraph.children[3], { italic: true });
+
+    expect(paragraph.children[4].text).toBe(' text');
+    checkMarks(paragraph.children[4], {});
+
+    expect(paragraph.children[5].text).toBe(' + some code');
+    checkMarks(paragraph.children[5], { code: true });
+  });
+
+  it('deserializes link tags', () => {
+    const deserialized = deserializeHtmlString(
+      '<p>paragraph <a href="https://abakus.no">with a link</a></p>'
+    );
+
+    expect(deserialized).toHaveLength(1);
+    const paragraph = deserialized[0];
+    expect(paragraph.type).toBe('paragraph');
+    expect(paragraph.children).toHaveLength(2);
+    expect(paragraph.children[0].text).toBe('paragraph ');
+    const link = paragraph.children[1];
+    expect(link.type).toBe('link');
+    expect(link.url).toBe('https://abakus.no');
+    expect(link.children).toHaveLength(1);
+    expect(link.children[0].text).toBe('with a link');
+  });
+
+  it('deserializes document with <br> tag', () => {
+    const deserialized = deserializeHtmlString(
+      '<p>paragraph <br> with a line break <br/> or two</p>'
```

Comment:
`<br>` (without a slash) is sufficient, no? Or did you specifically want to test it with and without a slash?

Reply:
I did want to test both, but now that I think about it it's very unnecessary as this is handled by the dom parser, and not the deserializer.